solver/app/reject.go

```diff
@@ -305,6 +311,36 @@ func checkLiquidity(ctx context.Context, expenses []TokenAmt, backend *ethbacken
 	return nil
 }
 
+// checkOrderCalls checks if all calls in an order are allowed.
+func checkOrderCalls(order Order, isAllowed callAllowFunc) error {
+	fill, err := order.ParsedFillOriginData()
+	if err != nil {
+		return errors.Wrap(err, "parse fill origin data")
+	}
+
+	var calls []types.Call
+	for _, call := range fill.Calls {
+		calls = append(calls, types.Call{
+			Target: call.Target,
+			Value:  call.Value,
+			Data:   append(call.Selector[:], call.Params...),
+		})
+	}
+
+	return checkCalls(order.DestinationChainID, calls, isAllowed)
+}
+
+// checkCalls checks if all calls to destChainID are allowed.
+func checkCalls(destChainID uint64, calls []types.Call, isAllowed callAllowFunc) error {
+	for _, call := range calls {
+		if !isAllowed(destChainID, call.Target, call.Data) {
+			return newRejection(rejectCallNotAllowed, errors.New("call not allowed", "target", call.Target.Hex(), "data", hexutil.Encode(call.Data)))
+		}
+	}
+
+	return nil
+}
```

Comment:
Should we add an error here (or catch and change before sending api response) that says something like "if you are a developer looking to use omni in your app, reach out to the team at some@email"?

Reply:
nah